app/frontend/containers/CallToolView/CallToolView.js

```diff
@@ -147,18 +148,14 @@ class CallToolView extends Component {
     return _.sample(candidates);
   }
 
-  selectedTargetIndex() {
-    return _.findIndex(this.props.targets, this.state.selectedTarget);
-  }
-
   submit(event: any) {
     event.preventDefault();
     if(!this.validateForm()) return;
     this.setState({ errors: {}, loading: true });
     ChampaignAPI.calls.create({
       pageId: this.props.pageId,
       memberPhoneNumber: this.state.form.memberPhoneCountryCode + this.state.form.memberPhoneNumber,
-      targetIndex: this.selectedTargetIndex()
+      targetId: this.state.selectedTarget.id
```

Comment:
Hey @eyko, thanks for the thorough explanation. What worries me about this last approach is that, even if in practice is very unlikely, logically there's a chance that the `then` callback is executed before the `this.setState({ loading: true });`, leaving the loading flag in the wrong state.